src/main/java/tutorlink/uipackage/Ui.java

```diff
@@ -0,0 +1,14 @@
+package tutorlink.uipackage;
+
+import tutorlink.exceptionspackage.MissingNextLineException;
+import java.util.Scanner;
+
+public class Ui {
+    private Scanner in = new Scanner(System.in);
+    public String collectUserInput() throws MissingNextLineException {
+        if(!in.hasNextLine()) {
+            throw new MissingNextLineException("No lines to parse!");
+        }
```

Comment:
I like that collectUserInput() checks for whether the user entered a blank line.
Perhaps moving forwards it could silently consume the blank line without considering it an exception?